tests/unit/test_bf16.cpp

```diff
@@ -0,0 +1,1101 @@
+#include "gtest/gtest.h"
+#include "VecSim/vec_sim.h"
+#include "VecSim/algorithms/hnsw/hnsw_single.h"
+#include "VecSim/index_factories/hnsw_factory.h"
+#include "test_utils.h"
+#include "VecSim/utils/serializer.h"
+#include "mock_thread_pool.h"
+#include "VecSim/query_result_definitions.h"
+#include "VecSim/types/bfloat16.h"
+#include "VecSim/vec_sim_debug.h"
+#include "VecSim/spaces/L2/L2.h"
+
+#include <random>
+
+using bfloat16 = vecsim_types::bfloat16;
+
+class BF16Test : public ::testing::Test {
+protected:
+    virtual void SetUp(HNSWParams &params) {
```

Comment:
Can't you just set it to be pure virtual? (=0)

Reply:
no because the argument type is different